packages/slate/src/interfaces/text.ts

```diff
@@ -111,7 +111,7 @@ export const Text = {
 
         // If the range starts after the leaf, or ends before it, continue.
         if (
-          start.offset > offset + length ||
+          start.offset >= offset + length ||
```

Comment:
@delijah @CameronAckermanSEL, it looks like collapsed decorators will never be rendered at the end of the line and intersected after this change.

Reply:
Thanks, this is a good catch